pkg/kubelet/cm/deviceplugin/endpoint.go

```diff
@@ -32,7 +32,15 @@ import (
 // endpoint maps to a single registered device plugin. It is responsible
 // for managing gRPC communications with the device plugin and caching
 // device states reported by the device plugin.
-type endpoint struct {
+type endpoint interface {
+	Run()
+	Stop()
+	Allocate(devs []string) (*pluginapi.AllocateResponse, error)
+	GetDevices() []pluginapi.Device
+	Callback(resourceName string, added, updated, deleted []pluginapi.Device)
```

Comment:
Why we need these functions to be public? Could them be called out of package `deviceplugin`?

Reply:
+1